official/mnist/dataset.py

```diff
@@ -89,15 +89,14 @@ def decode_image(image):
     image = tf.reshape(image, [784])
     return image / 255.0
 
-  def one_hot_label(label):
-    label = tf.decode_raw(label, tf.uint8)  # tf.string -> tf.uint8
-    label = tf.reshape(label, [])  # label is a scalar
-    return tf.one_hot(label, 10)
+  def decode_label(label):
+    label = tf.decode_raw(label, tf.uint8)  # tf.string -> [tf.uint8]
+    return tf.to_int32(label)
```

Comment:
Random thought: Would it make sense for the dataset to return a `uint8` tensor and push the casting to the point where it is required?  (The answer may well be no :))

Reply:
It's not big difference either way.

I add a commit to move the cast to the top of the model_fns.